integration-tests/smoke/ccip_batching_test.go

```diff
@@ -0,0 +1,295 @@
+package smoke
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/maps"
+
+	"github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+	"github.com/smartcontractkit/chainlink/deployment"
+	ccdeploy "github.com/smartcontractkit/chainlink/deployment/ccip"
+	"github.com/smartcontractkit/chainlink/integration-tests/ccip-tests/testsetups"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/multicall3"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/offramp"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/onramp"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/router"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+func Test_CCIPBatching(t *testing.T) {
+	// Setup 3 chains, with 2 lanes going to the dest.
+	lggr := logger.TestLogger(t)
+	ctx := ccdeploy.Context(t)
+	// Will load 3 chains when specified by the overrides.toml or env vars (E2E_TEST_SELECTED_NETWORK).
+	// See e2e-tests.yml.
+	e, _, _ := testsetups.NewLocalDevEnvironmentWithDefaultPrice(t, lggr)
+
+	state, err := ccdeploy.LoadOnchainState(e.Env)
+	require.NoError(t, err)
+
+	allChainSelectors := maps.Keys(e.Env.Chains)
+	require.Len(t, allChainSelectors, 3, "this test expects 3 chains")
+	sourceChain1 := allChainSelectors[0]
+	sourceChain2 := allChainSelectors[1]
+	destChain := allChainSelectors[2]
+	t.Log("All chain selectors:", allChainSelectors,
+		", home chain selector:", e.HomeChainSel,
+		", feed chain selector:", e.FeedChainSel,
+		", source chain selector 1:", sourceChain1,
+		", source chain selector 2:", sourceChain2,
+		", dest chain selector:", destChain,
+	)
+
+	// connect sourceChain1 and sourceChain2 to destChain
+	require.NoError(t, ccdeploy.AddLaneWithDefaultPrices(e.Env, state, sourceChain1, destChain))
+	require.NoError(t, ccdeploy.AddLaneWithDefaultPrices(e.Env, state, sourceChain2, destChain))
+
+	const (
+		numMessages = 5
```

Comment:
5 is very little, shouldn't we use some higher value? 20 or 30?

Reply:
If the delay caused by the sending ccip messages leads to fragmentation (creating multiple batches), you can try sending messages **before** proposing job specs. That way you "delay" the off-chain processing

Reply:
It shouldn't lead to fragmentation for commit because these messages are created in the same tx

Reply:
I can bump this to 30 and see what happens.